src/main/java/lotto/domain/Rank.java

```diff
@@ -0,0 +1,34 @@
+package lotto.domain;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+public enum Rank {
+    FIRST_PLACE(BigDecimal.valueOf(2000000000), (val, isBonus) -> val == 6),
```

Comment:
```suggestion
    FIRST_PLACE(2000000000, (val, isBonus) -> val == 6),
```

BigDecimal.valueOf의 반복을

```
Rank(final long prize, BiFunction<Integer, Boolean, Boolean> expression)
```

생성자를 통해 제거해볼 수 있을 것 같아요